src/lakefs_spec/spec.py

```diff
@@ -203,7 +208,10 @@ def get_file(
 
         super().get_file(rpath=rpath, lpath=lpath, callback=callback, outfile=outfile, **kwargs)
 
-    def info(self, path: str, **kwargs: Any) -> dict[str, Any]:
+    def info(
+        self, path: Union[str, os.PathLike[str], pathlib.Path], **kwargs: Any
+    ) -> dict[str, Any]:
+        path_as_str = stringify_path(path)
```

Comment:
Any reason you're not doing `path = stringify_path(path)`? Is mypy / ruff unhappy? (That would save the changed lines below.)

Reply:
Yep. That was it. MyPy does not resolve that `path = stringify_path(path)` is now a string.